### PR TITLE
Enhance display-test-app IMJS_USE_FRONTEND_TILES

### PR DESCRIPTION
**Pull request by DStradley**
The IMJS_USE_FRONTEND_TILES env var for DTA is changed to IMJS_FRONTEND_TILES_URL_TEMPLATE and is now a string instead of a boolean.  If set, it specifies the url to use.  In addition, the code that processes it was moved so that changes to the env var would take effect upon the next run of DTA.

The url string can also include special tokens: {iModel.key} or {iModel.filename}.  If present, these will get replaced by the value of iModel.key or just the filename of that (no path or extension), correspondingly.

E.g.: http://localhost:8080{iModel.key}-tiles/3dft/ or http://localhost:8080/MshX/{iModel.filename}/

Note that the contents of iModel.key will be different on different OSes.

### CODE REVIEW

File: test-apps/display-test-app/README.md [hunk 1]
```diff
@@ -216,7 +216,9 @@ You can use these environment variables to alter the default behavior of various
 * IMJS_NO_ELECTRON_AUTH
   * If defined, the authorization client will not be initialized for the electron app, to work around a current bug that causes it to produce constant exceptions when attempting to obtain an access token.
 * IMJS_USE_FRONTEND_TILES
-  * If defined, the @itwin/frontend-tiles package will be used to obtain tile trees for spatial views, served over localhost.
+  * If defined, specifies url for @itwin/frontend-tiles to obtain tile trees for spatial views, served over localhost.  The string can include special
```
Comment (pmconne) [inline test-apps/display-test-app/README.md:219]: Please change name of env var to reflect that it is a URL template, not a boolean.

File: test-apps/display-test-app/src/common/DtaConfiguration.ts [hunk 2]
```diff
@@ -55,6 +54,7 @@ export interface DtaStringConfiguration {
   oidcClientId?: string; // default is undefined, used for auth setup
   oidcScope?: string; // default is undefined, used for auth setup
   oidcRedirectUri?: string; // default is undefined, used for auth setup
+  useFrontendTiles?: string; // if set, specifies url for @itwin/frontend-tiles to obtain tile trees for spatial views.  See README.md
```
Comment (pmconne) [inline test-apps/display-test-app/src/common/DtaConfiguration.ts:57]: Rename to `frontendTilesUrlTemplate` or something similarly descriptive.

File: test-apps/display-test-app/src/frontend/DisplayTestApp.ts [hunk 3]
```diff
@@ -146,6 +146,18 @@ function setConfigurationResults(): [renderSystemOptions: RenderSystem.Options,
   return [renderSystemOptions, tileAdminProps];
 }
 
+// simple function to extract file name, without path or extension, on Windows or Linux
+function getFileName(path: string): string {
```
Comment (pmconne) [inline test-apps/display-test-app/src/frontend/DisplayTestApp.ts:150]: Seems like `path` ought to provide a more straightforward cross-platform way to do this - did you look?

**Review by pmconne — CHANGES_REQUESTED** [review]
see comments

File: test-apps/display-test-app/src/frontend/DisplayTestApp.ts (hunk 3 above)
Comment (pmconne) [reply test-apps/display-test-app/src/frontend/DisplayTestApp.ts:150]: Oh, right. Forget I said anything.

File: test-apps/display-test-app/README.md (hunk 1 above)
Comment (DStradley) [reply test-apps/display-test-app/README.md]: Done

File: test-apps/display-test-app/src/common/DtaConfiguration.ts (hunk 2 above)
Comment (DStradley) [reply test-apps/display-test-app/src/common/DtaConfiguration.ts]: Done

**Review by pmconne — APPROVED** [review]
`{iModel.filename}` seems to work fine with a unix path.

**Review by markschlosseratbentley — APPROVED** [review]

**Pull request merged**